numba/np/arraymath.py

```diff
@@ -376,11 +376,14 @@ def array_prod_impl(a):
 @overload_method(types.Array, "cumsum")
 def array_cumsum(a):
     if isinstance(a, types.Array):
-        is_integer = a.dtype in types.signed_domain
+        is_signed_integer = a.dtype in types.signed_domain
+        is_unsigned_integer = a.dtype in types.unsigned_domain
         is_bool = a.dtype == types.bool_
-        if (is_integer and a.dtype.bitwidth < types.intp.bitwidth)\
+        if (is_signed_integer and a.dtype.bitwidth < types.int_.bitwidth)\
```

Comment:
Shouldn't this be `intp`?

```suggestion
        if (is_signed_integer and a.dtype.bitwidth < types.intp.bitwidth)\
```